dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Internals/IPicklerImplementation.cs

```diff
@@ -0,0 +1,15 @@
+﻿/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */
+
+using System;
+
+namespace Razorvine.Pickle
+{
+    internal interface IPicklerImplementation : IDisposable
```

Comment:
Is this interface required? I only see 1 class implement it, and we don't have the analogous `IUnpicklerImplementation`.

Reply:
the `Pickler` implements a `save` method which allows writing to the stream provided in the `dump` method

```cs
var stream = new MemoryStream();
var pickler = new Pickler();
pickler.dump(object_1, stream);
pickler.save(object_2); // writes the object_2 to stream provided in dump
```

this is why I have to store the reference to `PicklerImplementation`. The problem is that  the type is a generic struct (there is no base class) and to be able to save it in a field I would either have to make the Pickler generic (breaking change) or introduce this interface.

```cs
class Pickler
{
       private IPicklerImplementation _picklerImplementation;

       public void dumps(object o, ref byte[] output, out int bytesWritten)
           => _picklerImplementation = new PicklerImplementation<ArrayWriter>();
        
       public void dump(object o, Stream stream)
            => _picklerImplementation = new PicklerImplementation<StreamWriter>();
 
       public void save(object o) => _picklerImplementation.save(o);
}
```

Reply:
Ah, I got it. Makes sense.